docs/content/users/install/shell-completion.md

```diff
@@ -2,66 +2,87 @@
+
+When you install DDEV via Homebrew, each new release will automatically get a refreshed completions script.
+
+## macOS Zsh + Homebrew
+
+To make Homebrew completions available in Zsh the Homebrew-managed path `zsh/site-functions` has to be added to the FPATH. There are two scenarios:
+
+=== "macOS Zsh"
+
+    Add the following block to your `~/.zshrc` file:
+
+    ```bash
+    if type brew &>/dev/null
+    then
+      FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"
+      autoload -Uz compinit
+      compinit
+    fi
     ```
 
-    !!!note "Bash profile"
-        You must add the include to your `.bash_profile` or `.profile` or nothing will work. Use `source ~/.bash_profile` or `source ~/.profile` to make it take effect immediately.
+    Note that the sourcing of the FPATH has to happen before the Zsh completion index is initialized with `compinit`.
 
-    Link completions by running `brew completions link`.
+=== "macOS Zsh with Oh My Zsh"
 
-    When you install DDEV via Homebrew, each new release will automatically get a refreshed completions script.
+    Oh-My-Zsh is calling `compinit` for you when `oh-my-zsh.sh` is sourced. Instead of adding the block that was necessary for `Plain macOS Zsh` place the following line right before `oh-my-zsh.sh` is sourced:
+
+    ```bash
+    FPATH="$(brew --prefix)/share/zsh/site-functions:${FPATH}"
+    ```
 
-=== "Bash/Zsh/Fish on Linux"
+To avoid any potential caching issue forcibly rebuild the `.zcompdump` file, which is the index for Zsh completions:
 
-    ## Bash/Zsh/Fish on Linux
+```bash
+rm -f ~/.zcompdump; compinit
+```
 
-    On Debian and Yum based systems, if you installed DDEV using `apt install ddev`, the `bash`, `zsh`, and `fish` completions should be automatically installed at `/usr/share/bash-completion/completions/ddev`, `/usr/share/zsh/vendor-completions/_ddev` and `/usr/share/fish/completions/ddev.fish` respectively, and the `bash` completions should be automatically installed at `/usr/share/bash-completion/completions/bash`.
+In case you run into any `zsh compinit: insecure directories` warnings, simply run:
 
-    Otherwise, you can download the completion files for manual installation as described [below](#tar-archive-of-completion-scripts-for-manual-deployment). Every Linux distro requires a different manual installation technique. On Debian/Ubuntu, you could deploy the `ddev_bash_completion.sh` script where it needs to be by running `sudo mkdir -p /usr/share/bash-completion/completions && sudo cp ddev_bash_completion.sh /usr/share/bash-completion/completions/ddev`.
+```bash
+chmod -R go-w "$(brew --prefix)/share"
+```
 
-=== "Oh-My-Zsh"
+Otherwise, you can extract the Zsh completions file (`ddev_zsh_completion.sh`) from the tar archive of completion scripts included with each release. Read more in the [tar archive of completion scripts for manual deployment](#tar-archive-of-completion-scripts-for-manual-deployment) section.
```

Comment:
I think this paragraph belongs elsewhere, wherever we describe all the many other possibililties, rather than here.

Reply:
technically it has nothing to do with macos zsh with homebrew. i've already striked a part but kept that part. but better do a clear separation of concern here.